src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs

```diff
@@ -33,6 +36,13 @@ public static IApplicationBuilder UseWebJobsScriptHost(this IApplicationBuilder
                 builder.UseMiddleware<PlaceholderSpecializationMiddleware>();
             }
 
+            // Specialization can change the CompatMode setting, so this must run later than
+            // the PlaceholderSpecializationMiddleware
+            builder.UseWhen(context => v2CompatOptions.CurrentValue.IsV2CompatibilityModeEnabled || FeatureFlags.IsEnabled(ScriptConstants.FeatureFlagAllowSynchronousIO), config =>
```

Comment:
This will be re-evaluated often. Would be good to avoid the lookup and processing that we perform on the feature flags (i.e. cache the results) to avoid the subsequent hits.

Reply:
Ah, right. Should we have a FeatureFlagsOptions or something? Seems like overkill to have one for each flag.

Reply:
I don’t think we need to go that far, and that wouldn’t be required for every flag.